### PR TITLE
Styled component

### PR DESCRIPTION
**Pull request by umangag07**
Made the components with the styled-components library and removed the .css file components

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1374,26 +1358,22 @@
...
       "resolved": "https://registry.npmjs.org/@emotion/utils/-/utils-0.11.3.tgz",
-      "integrity": "sha512-0o4l6pZC+hI88+bzuaX/6BgOvQVhbt2PfmxauVaYOGgbsAw14wdKyvMCZXnsnsHys94iadcF+RG/wZyx6+ZZBw==",
-      "dev": true
+      "integrity": "sha512-0o4l6pZC+hI88+bzuaX/6BgOvQVhbt2PfmxauVaYOGgbsAw14wdKyvMCZXnsnsHys94iadcF+RG/wZyx6+ZZBw=="
     },
     "@emotion/weak-memoize": {
       "version": "0.2.5",
       "resolved": "https://registry.npmjs.org/@emotion/weak-memoize/-/weak-memoize-0.2.5.tgz",
-      "integrity": "sha512-6U71C2Wp7r5XtFtQzYrW5iKFT67OixrSxjI4MptCHzdSVlgabczzqLe0ZSgnub/5Kp4hSbpDB1tMytZY9pwxxA==",
-      "dev": true
```
Comment (digvijayu) [inline package-lock.json:1396]: Looks like you did something with the dependencies.. please remove node modules and update again

File: src/Component/BoxComponent/Box.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+export interface BoxProps {
+  id: string;
+  children: ReactNode;
+  size: String;
+};
+export const BoxComp = styled('div')<BoxProps>`
+  background-color:${props=>
+    props.id==="primary-box"?'#00b4d8':
```
Comment (digvijayu) [inline src/Component/BoxComponent/Box.tsx]: please define color as constants and import them here

File: src/Component/BoxComponent/Box.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    props.id==="success-box"?'#52b788':
+    props.id==="warning-box"?'#ffb703':'#fff'
+  };
+  height:${props=>
+    props.size==="small"?'10rem':
+    props.size==="medium"?'30rem':
+    props.size==="large"?'50rem':'10rem'
+  };
+  width:${props=>
+    props.size==="small"?'10rem':
```
Comment (digvijayu) [inline src/Component/BoxComponent/Box.tsx]: please add standard size constants and import them

File: src/Component/Heading/Heading.tsx [hunk 4]
```diff
@@ -1,13 +1,36 @@
...
 export interface HeadingProp {
-  type: String;
+  id: string;
   children: ReactNode;
   size: String;
 }
+export const HeadingComp = styled('div')<HeadingProp>`
+background-color:${props=>
+  props.id==="primary-heading"?'#00b4d8':
+    props.id==="secondary-heading"?'#03045e':
```
Comment (digvijayu) [inline src/Component/Heading/Heading.tsx]: same as above

**Review by digvijayu — CHANGES_REQUESTED** [review]

File: src/Component/BoxComponent/Box.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    if (props.id === "danger-box") {
+      return color.danger;
+    }
+    if (props.id === "success-box") {
+      return color.success;
+    }
+    if (props.id === "warning-box") {
+      return color.warning;
+    }
+    return color.primary;
```
Comment (digvijayu) [inline src/Component/BoxComponent/Box.tsx]: Might make sense to use `switch`

File: src/Component/BoxComponent/Box.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    if (props.size === "small") {
+      return boxHW.height.small;
+    } if (props.size === "medium") {
+      return boxHW.height.medium;
+    }
+    if (props.size === "large") {
+      return boxHW.height.large;
+    }
+
+    return boxHW.height.medium;
```
Comment (digvijayu) [inline src/Component/BoxComponent/Box.tsx:39]: use of `switch` feels more appropriate

File: src/Component/BoxComponent/Box.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    if (props.size === "small") {
+      return boxHW.width.small;
+    } if (props.size === "medium") {
+      return boxHW.width.medium;
+    }
+    if (props.size === "large") {
+      return boxHW.width.large;
+    }
+
+    return boxHW.width.medium;
```
Comment (digvijayu) [inline src/Component/BoxComponent/Box.tsx:54]: Pls remove the width it should be automatic

File: src/Component/BoxComponent/Box.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+  }
+
+  };
+  color:white;
```
Comment (digvijayu) [inline src/Component/BoxComponent/Box.tsx]: Pls use constants

File: src/Component/Button/Button.tsx [hunk 9]
```diff
@@ -1,26 +1,76 @@
...
+    if (props.id === "danger-button") {
+      return color.danger;
+    }
+    if (props.id === "success-button") {
+      return color.success;
+    }
+    if (props.id === "warning-button") {
+      return color.warning;
+    }
+    return color.primary;
```
Comment (digvijayu) [inline src/Component/Button/Button.tsx]: Pls use `switch`

File: src/Component/Button/Button.tsx [hunk 10]
```diff
@@ -1,26 +1,76 @@
...
+    if (props.size === "small") {
+      return buttonHW.height.small;
+    } if (props.size === "medium") {
+      return buttonHW.height.medium;
+    }
+    if (props.size === "large") {
+      return buttonHW.height.large;
+    }
+
+    return buttonHW.height.medium;
```
Comment (digvijayu) [inline src/Component/Button/Button.tsx:41]: Pls use `switch`

File: src/Component/Button/Button.tsx [hunk 11]
```diff
@@ -1,26 +1,76 @@
...
+      return buttonHW.width.small;
+    } if (props.size === "medium") {
+      return buttonHW.width.medium;
+    }
+    if (props.size === "large") {
+      return buttonHW.width.large;
+    }
+
+    return buttonHW.width.medium;
+
```
Comment (digvijayu) [inline src/Component/Button/Button.tsx:55]: Pls use `switch`

File: src/Component/Button/Button.tsx [hunk 12]
```diff
@@ -1,26 +1,76 @@
...
+  }
+  };
+  font-size: 1rem;
```
Comment (digvijayu) [inline src/Component/Button/Button.tsx]:
```suggestion
  font-size: ${FONT_SIZE.M};
```

File: src/Component/Button/Button.tsx [hunk 13]
```diff
@@ -1,26 +1,76 @@
...
+  color: white;
+  text-align: center;
+  padding: 0.2rem;
```
Comment (digvijayu) [inline src/Component/Button/Button.tsx]:
```suggestion
  padding: ${SPACE.XS};
```

File: src/styles.tsx [hunk 14]
```diff
@@ -0,0 +1,43 @@
+export const color={
```
Comment (digvijayu) [inline src/styles.tsx]:
```suggestion
export const COLOR={
```

File: src/styles.tsx [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+        small:"10rem",
+        medium:"30rem",
+        large:"50rem"
+      },
+      width:{
+        small:"10rem",
+        medium:"30rem",
+        large:"50rem"
+      }
+};
```
Comment (digvijayu) [inline src/styles.tsx:19]: Box height and width should not be set directly. Allow the components to take the natural space. Managing responsiveness is easier when size is not set

File: src/styles.tsx [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+        small:"2rem",
+        medium:"3rem",
+        large:"5rem"
+      },
+      width:{
+        small:"10rem",
+        medium:"15rem",
+        large:"20rem"
+      }
+};
```
Comment (digvijayu) [inline src/styles.tsx]:
```suggestion
export const SPACE = {
  XXS: '0.2rem',
  XS: '0.4rem',
  S: '0.8rem',
  M: '1rem',
  L: ...,
  XL: ...,
  XXL...
}
```

Comment (digvijayu) [reply src/styles.tsx]: same space can be used in padding, margin etc

**Review by digvijayu — CHANGES_REQUESTED** [review]

File: src/Component/BoxComponent/Box.tsx (hunk 5 above)
Comment (umangag07) [reply src/Component/BoxComponent/Box.tsx]: okay, I will change that.

File: src/Component/BoxComponent/Box.tsx (hunk 6 above)
Comment (umangag07) [reply src/Component/BoxComponent/Box.tsx:39]: yes, I forgot about the switch.

File: src/styles.tsx (hunk 16 above)
Comment (umangag07) [reply src/styles.tsx]: yup, I will make space constant.

File: src/styles.tsx (hunk 15 above)
Comment (umangag07) [reply src/styles.tsx:19]: This was a demo component, and to try the storybook things I made the height and width static.

Comment (umangag07) [reply src/styles.tsx:19]: But for our main components, we will make it responsive.

**Review by digvijayu — APPROVED** [review]

**Pull request merged**